HACKING.md

```diff
@@ -42,73 +53,135 @@ Add `$GOPATH/bin` to your `PATH`, so you can run the go programs you install:
-that dependencies for your distribution are installed. For example, on Ubuntu,
-run:
+Build dependencies can automatically be resolved using `build-deb`on Ubuntu.
 
-    sudo apt-get build-dep ./
+    cd ~/snapd
+    sudo apt-get build-dep .
 
-### Building
+Package build dependencies for other distributions can be found under the
+`packages\`directory.
 
-To build, once the sources are available and `GOPATH` is set, you can just run
+Go module dependencies are automatically resolved at build time.
 
-    go build -o /tmp/snap github.com/snapcore/snapd/cmd/snap
+### Building (natively)
 
-to get the `snap` binary in /tmp (or without -o to get it in the current
-working directory). Alternatively:
+To build the `snap` commandline client:
 
-    go install github.com/snapcore/snapd/cmd/snap/...
+```
+cd ~/snapd
+mkdir -p /tmp/build
+go build -o /tmp/build/snapd ./cmd/snapd
+```
 
-to have it available in `$GOPATH/bin`
+To build the `snapd` REST API daemon:
 
-Similarly, to build the `snapd` REST API daemon, you can run
+```
+cd ~/snapd
+mkdir -p /tmp/build
+go build -o /tmp/build/snapd ./cmd/snapd
+```
 
-    go build -o /tmp/snapd github.com/snapcore/snapd/cmd/snapd
+To build the all`snapd` Go components:
 
-### Contributing
+```
+cd ~/snapd
+mkdir -p /tmp/build
+go build -o /tmp/build ./...
+```
 
-Contributions are always welcome! Please make sure that you sign the
-Canonical contributor license agreement at
-http://www.ubuntu.com/legal/contributors
+### Cross-compiling (example: ARM target)
 
-Snapd can be found on GitHub, so in order to fork the source and contribute,
-go to https://github.com/snapcore/snapd. Check out [GitHub's help
-pages](https://help.github.com/) to find out how to set up your local branch,
-commit changes and create pull requests.
+Install a suitable cross-compiler for the target architecture.
 
-We value good tests, so when you fix a bug or add a new feature we highly
-encourage you to create a test in `$source_test.go`. See also the section
-about Testing.
+```
+sudo apt-get install gcc-arm-linux-gnueabihf
+```
+
+`Snapd` depends on libseccomp v2.3 or later. The following instructions can be used to
+cross-compile the library:
+
+```
+cd ~/
+git clone https://github.com/seccomp/libseccomp
+cd libseccomp
+./autogen.sh
+./configure --host=arm-linux-gnueabihf --prefix=${HOME}/libseccomp/build
+make && make install
+```
+
+Setup the Go environment for cross-compiling.
+
+```
+export CC=arm-linux-gnueabihf-gcc
+export CGO_ENABLED=1
+export CGO_LDFLAGS="-L${HOME}/libseccomp/build/lib"
+export GOOS=linux
+export GOARCH=arm
+```
+
+To build the `snap` commandline client:
+
+```
+cd ~/snapd
+mkdir -p /tmp/build
+go build -o /tmp/build/snapd ./cmd/snapd
+```
+
+To build the `snapd` REST API daemon:
+
+```
+cd ~/snapd
+mkdir -p /tmp/build
+go build -o /tmp/build/snapd ./cmd/snapd
+```
+
+To build all`snapd` Go components:
+
+```
+cd ~/snapd
+mkdir -p /tmp/build
+go build -o /tmp/build ./...
+```
```

Comment:
Since there are no major differences between this and the previous building instructions, maybe this can be shortened to just a reference to the previous section? Alternatively, we could just leave the last block, I think it would still be clear that you can build snap and snapd in the same way